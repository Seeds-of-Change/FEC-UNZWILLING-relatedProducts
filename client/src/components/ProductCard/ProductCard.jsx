import React from 'react';
import PropTypes from 'prop-types';
import StarRating from '../StarRating/StarRating';
import style from './styles.css';

export default class ProductCard extends React.Component {
  constructor(props) {
    super(props);

    const {
      id,
      image,
      imageMini,
      isFavorite,
      name,
      onSale,
      price,
      rating,
      related_products,
      salePrice,
      brand,
    } = props.product;

    // console.log(props);
    this.state = {
      id,
      image,
      imageMini,
      isFavorite,
      name,
      onSale,
      price,
      rating,
      related_products,
      salePrice,
      brand,
    };

    this.toggleFavorite = this.toggleFavorite.bind(this);
  }

  toggleFavorite(e) {
    const { updateProduct } = this.props;

    // remove this once API is working
    this.setState((prevState) => ({ isFavorite: !prevState.isFavorite }));

    // use this when API is working
    // const updatedProduct = this.state;
    // updatedProduct.isFavorite = !updatedProduct.isFavorite;
    // updateProduct(updatedProduct);
  }

  render() {
    const {
      isFavorite, brand, name, imageMini, rating,
    } = this.state;
    return (
      <div className={style.container}>
        <div className={style.stage}>
          <button type="button" className={isFavorite ? `${style.heart} ${style.is_active}` : style.heart} onClick={this.toggleFavorite}>{ }</button>
        </div>
        <img className={style.image} src={imageMini} alt="thumbnail" />
        <div className={style.brand}>{brand}</div>
        <div className={style.name}>{name}</div>
        <StarRating rating={rating} />
      </div>
    );
  }
}

// Carousel.defaultProps = {
//   header: null,
//   features: null,
//   updateActive: null,
//   activeIndex: 0,
// };

// Carousel.propTypes = {
//   features: PropTypes.arrayOf(
//     PropTypes.shape({
//       heading: PropTypes.string,
//       description: PropTypes.string,
//       posX: PropTypes.number,
//       posY: PropTypes.number,
//     }),
//   ),
//   header: PropTypes.string,
//   updateActive: PropTypes.func,
//   activeIndex: PropTypes.number,
// };
