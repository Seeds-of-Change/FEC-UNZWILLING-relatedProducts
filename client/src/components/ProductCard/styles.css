.container {
  height: 475px;
  width: 157px;
  padding: 3px 6px 0 4px;
  margin: 0 15px;
  display:inline-block;
  background:white;
  vertical-align: top;
  white-space: normal;
  position:relative;
}

/* // TWITTER HEART */
button.heart {
  border: none;
  padding: 0;
  outline:inherit;
  width: 50px;
  height: 50px;
  background: url("https://cssanimation.rocks/images/posts/steps/heart.png") no-repeat;
  background-position: 0.8% 50%;
  cursor: pointer;
  transition: background-position 1s steps(28);
  transition-duration: 0s;
}

.heart.is_active {
  transition-duration: 1s;
  background-position: 99% 50%;
}

.stage {
  position: absolute;
  top: 5%;
  right: 15%;
  transform: translate(50%, -50%);
}

.image {
  display: inline-block;
  vertical-align: middle;
  width: auto;
  margin: 50px 0;
  max-width: 100%;
}

.brand, .name {
  text-align: left;
  padding: 0 8px;
  overflow: hidden;
  margin-bottom: 0;
}
.brand {
  /* font-family: futura book; */
  color: #000;
  font-size: .875rem;
  text-transform: uppercase;
  text-align: left;
  letter-spacing: .02em;
  max-height: 32px;
  overflow: hidden;
}

.name {
  /* font-family: futura bold; */
  color: #27221d;
  font-size: .875rem;
  font-weight: 700;
  text-transform: uppercase;
  line-height: 21px;
  letter-spacing: .1px;
  min-height: 20px;
  line-height: 21px;
  padding-top: 3px;
  max-height: 125px;
}

.prices {
  position: absolute;
  width: 100%;
  vertical-align: bottom;
  overflow: hidden;
  bottom: 5px;
  right: 5px;
  text-align: right;
  color: #27221d;
}

.oldPrice {
  /* font-family: futura book; */
  font-size: .6875rem;
  line-height: 18px;
  letter-spacing: 1px;
  text-decoration: line-through;
}

.price {
  /* font-family: futura bold; */
  font-size: .875rem;
  text-transform: uppercase;
  line-height: 18px;
  letter-spacing: 1px;
}