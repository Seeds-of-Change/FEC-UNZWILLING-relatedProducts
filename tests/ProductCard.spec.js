import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard from '../client/src/components/ProductCard/ProductCard';

describe('ProductCard', () => {
  describe('Have props', () => {
    let component;

    beforeEach(() => {
      const { product } = {
        id: 1,
        brand: 'Zwilling ENFINIGY',
        name: 'Power Blender',
        product_features: {},
        related_products: [1, 3, 15, 21],
        image: 'https://unzwilling.s3-us-west-1.amazonaws.com/1.jpg',
        imageMini: 'https://unzwilling.s3-us-west-1.amazonaws.com/1_mini.jpg',
        price: 100,
        salePrice: 80,
        onSale: true,
        rating: 4.5,
        isFavorite: true,
      };

      component = renderer.create(
        <ProductCard product={product} />,
      );
    });

    test('snapshot renders', () => {
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
