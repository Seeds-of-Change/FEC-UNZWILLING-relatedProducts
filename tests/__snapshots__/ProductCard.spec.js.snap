// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Have props snapshot renders 1`] = `
<div
  className="container"
>
  <div
    className="stage"
  >
    <button
      className="heart"
      onClick={[Function]}
      type="button"
    />
  </div>
  <img
    alt="thumbnail"
    className="image"
  />
  <div
    className="brand"
  />
  <div
    className="name"
  />
  <div
    className="star_ratings"
  >
    <div
      className="fill_ratings"
      style={
        Object {
          "width": "0%",
        }
      }
    >
      <span>
        ★★★★★
      </span>
    </div>
    <div
      className="empty_ratings"
    >
      <span>
        ★★★★★
      </span>
    </div>
  </div>
  <div
    className="prices"
  >
    <div
      className="price"
    >
      $
    </div>
  </div>
</div>
`;
